File: dart/word-count/lib/word_count.dart
class WordCount {
  RegExp nonAlpha = RegExp(r"[^a-zA-Z0-9'\s]");
  RegExp leadingTrailingApostrophes = RegExp(r"\b'\b|\b'\b");

  // Put your code here
  Map<String, int> countWords(String subtitles) {
    subtitles = subtitles.toLowerCase();
    // subtitles = subtitles.replaceAll(nonAlpha, '');
    // subtitles = subtitles.replaceAll(leadingTrailingApostrophes, '');

    final words = subtitles.split(RegExp(r"[^\w'\w]"));

    print(subtitles);
    print(words);

    final wordCount = <String, int>{};

    for (var word in words) {
      if (wordCount.containsKey(word)) {
        wordCount[word] = wordCount[word]! + 1;
      } else {
        wordCount[word] = 1;
      }
    }

    return wordCount;
  }
}

File: javascript/word-count/word-count.js
//
// This is only a SKELETON file for the 'Word Count' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const countWords = (phrase) => {
	phrase = phrase.toLowerCase();

	const filterRegex = /[^\w']/gi;

	phrase = phrase.replace(filterRegex, " ");

	const apostropheFilterRegex =
		/(?<!\w)\'(?!\w)|(?<=\w)\'(?!\w)|(?<!\w)\'(?=\w)/gi;

	phrase = phrase.replace(apostropheFilterRegex, " ");

	return phrase
		.split(" ")
		.filter(Boolean)
		.reduce((acc, cur) => ({ ...acc, [cur]: ~~acc[cur] + 1 }), {});
};
